cmd/spire-server/util/util.go

```diff
@@ -37,3 +50,86 @@ func Pluralizer(msg string, singular string, plural string, val int) string {
 
 	return result
 }
+
+type Clients struct {
+	AgentClient agent.AgentClient
```

Comment:
Quetion: If we want another API client,  add member in this struct?
e.g.
```go

type Clients struct {
	AgentClient agent.AgentClient
        SVIDClient  svid.SVIDClient
        ...
}
```

If so, NewClients() function always prepare all API clients?

Reply:
a downside of that approach is that we may create all clients when needing only one, how about change it to keep a connection and then we can have methods to just create new clients based on that, and we need a Release() funciton too:
```
type ServerClient interface {
   Release()
   NewAgentClient() agent.AgentClient 
   ...
}
type serverClient struct {
   conn *grpc.ClientConn
}
func NewServerClient(socket string) (*ServerClient, err) {
   conn, err := grpc.Dial(socketPath, grpc.WithInsecure(), grpc.WithDialer(dialer))
   if err != nil {
     return nil, err
   }  
   return &ServerClient{ conn: conn}, nil
}

func (c *serverClient) Release() {
    c.conn.Close()
}
func (c *serverClient) NewAgentClient() agent.AgentClient {
   return agent.NewAgentClient(c.conn)
}
```
What do you think about it?